### PR TITLE
Vue: Add repro template for vue-cli

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

- [x] Added repro template for vue-cli (Vue3)
- [x] Added repro template for vue-cli (Vue2) but commented out due to https://github.com/storybookjs/storybook/issues/19204
- [x] Fixed a bug with TS handling in Vue3
- [x] Added the ability to skip certain play functions based on which renderer you're in
- [x] Prefixed chromatic junit reports with the template-key to make failures easier to debug

## How to test

- [ ] CI passes except for existing errors on the `next` branch

### CODE REVIEW

**Comment by shilman** [comment]
**Vue2 args handling.** Our framework-independent stories use a pattern where we override args to the `Pre` component, e.g.

```js
  decorators: [
    (storyFn: PartialStoryFn, context: StoryContext) =>
      storyFn({ args: { object: context.argTypes } }),
  ],
```

This does not work with Vue2 args handling. We should change the way Vue2 rendering works to make it consistent with the other frameworks.

File: code/addons/interactions/src/preset/preview.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ const addSpies = (id: string, val: any, key?: string): any => {
     if (Array.isArray(val)) {
       return val.map((item, index) => addSpies(id, item, `${key}[${index}]`));
     }
-    if (typeof val === 'function' && val.name === 'actionHandler') {
+    if (typeof val === 'function' && val.isAction) {
```
Comment (tmeasday) [inline code/addons/interactions/src/preset/preview.ts:40]: We should get @ghengeveld to check this change.

File: scripts/tasks/chromatic.ts [hunk 2]
```diff
@@ -10,14 +11,22 @@ export const chromatic: Task = {
...
+          --projectToken=${token}`,
+        { cwd: sandboxDir },
+        { debug: true }
+      );
+    } finally {
+      if (fs.existsSync(junitFilename)) {
+        const junitXml = await (await fs.readFile(junitFilename)).toString();
+        const prefixedXml = junitXml.replace(/classname="(.*)"/g, `classname="${templateKey} $1"`);
+        await fs.writeFile(junitFilename, prefixedXml);
+      }
```
Comment (tmeasday) [inline scripts/tasks/chromatic.ts:29]: Can we pull this snippet up and apply it to the test runner and e2e tasks also?

**Review by tmeasday — APPROVED** [review]
LGTM! Like the skipping approach.

**Comment by shilman** [comment]
@tmeasday I've split of the Vue2 changes into a separate PR that needs more work. https://github.com/storybookjs/storybook/pull/19207

File: code/lib/core-server/src/__snapshots__/vue-3-cli_preview-dev-posix [hunk 3]
```diff
@@ -40,7 +40,7 @@ Object {
         "include": Array [
           "ROOT",
         ],
-        "test": "/\\\\.(mjs|tsx?|jsx?)$/",
```
Comment (yannbf) [inline code/lib/core-server/src/__snapshots__/vue-3-cli_preview-dev-posix:43]: I know this is just a snapshot update, but was the removal of tsx intentional?

File: code/addons/interactions/src/preset/preview.ts (hunk 1 above)
Comment (ghengeveld) [reply code/addons/interactions/src/preset/preview.ts:40]: How about either/or?

**Pull request merged**